internal/provisioner/certmanager.go

```diff
@@ -1,164 +0,0 @@
-package provisioner
-
-import (
-	"context"
-	"fmt"
-	"strings"
-	"time"
-
-	"github.com/mattermost/mattermost-cloud/internal/tools/k8s"
-	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
-	"github.com/mattermost/mattermost-cloud/internal/tools/utils"
-	"github.com/mattermost/mattermost-cloud/model"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
-)
-
-type certManager struct {
-	provisioner    *KopsProvisioner
-	kops           *kops.Cmd
-	logger         log.FieldLogger
-	desiredVersion string
-	actualVersion  string
-}
-
-func newCertManagerHandle(desiredVersion string, provisioner *KopsProvisioner, kops *kops.Cmd, logger log.FieldLogger) (*certManager, error) {
-	if logger == nil {
-		return nil, errors.New("cannot instantiate Cert Manager handle with nil logger")
-	}
-
-	if provisioner == nil {
-		return nil, errors.New("cannot create a connection to Cert Manager if the provisioner provided is nil")
-	}
-
-	if kops == nil {
-		return nil, errors.New("cannot create a connection to Cert Manager if the Kops command provided is nil")
-	}
-
-	return &certManager{
-		provisioner:    provisioner,
-		kops:           kops,
-		logger:         logger.WithField("cluster-utility", model.CertManagerCanonicalName),
-		desiredVersion: desiredVersion,
-	}, nil
-
-}
-
-func (n *certManager) updateVersion(h *helmDeployment) error {
-	actualVersion, err := h.Version()
-	if err != nil {
-		return err
-	}
-
-	n.actualVersion = actualVersion
-	return nil
-}
-
-func (n *certManager) CreateOrUpgrade() error {
-	h := n.NewHelmDeployment()
-	err := h.Update()
-	if err != nil {
-		return err
-	}
-
-	err = n.updateVersion(h)
-	return err
-}
-
-func (n *certManager) DesiredVersion() string {
-	return n.desiredVersion
-}
-
-func (n *certManager) ActualVersion() string {
-	return strings.TrimPrefix(n.actualVersion, "cert-manager-")
-}
-
-func (n *certManager) Destroy() error {
```

Comment:
It would be nice if we performed cleanup of this utility. This is where it normally would happen. The idea here is we could perform a `cluster provision` to cleanup old utilities.

Reply:
you mean, we clean that and later we remove the cert-manager code? 
I can revert this to make that
i was planning to remove manually since is not many clusters to perform that

Reply:
Yeah, this probably won't be the first time we add and remove a utility, so I think we should get used to having the provisioner cleanup stuff we no longer need. We should be able to run `provision` on the cluster and have the new ACM stuff install first and then remove the cert manager resources.